float32 x
float32 y
float32 mass
uint8 color